## Decodelabs Internship Project

## Project 1 
## Data Cleaning and Preprocessing

In [1]:
# import libraries

import pandas as pd, numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# load dataset

df =pd.read_csv(r"C:\Users\HP\Documents\Dataset for Data Analytics.csv")
df.head()

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


In [3]:
#  check the sturcture of the data

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   OrderID          1200 non-null   object 
 1   Date             1200 non-null   object 
 2   CustomerID       1200 non-null   object 
 3   Product          1200 non-null   object 
 4   Quantity         1200 non-null   int64  
 5   UnitPrice        1200 non-null   float64
 6   ShippingAddress  1200 non-null   object 
 7   PaymentMethod    1200 non-null   object 
 8   OrderStatus      1200 non-null   object 
 9   TrackingNumber   1200 non-null   object 
 10  ItemsInCart      1200 non-null   int64  
 11  CouponCode       891 non-null    object 
 12  ReferralSource   1200 non-null   object 
 13  TotalPrice       1200 non-null   float64
dtypes: float64(2), int64(2), object(10)
memory usage: 131.4+ KB


In [4]:
# check for the shape and features of the data 

df.shape

(1200, 14)

There are 1200 total number customers observed and 14 features (column)

In [5]:
# check for missing value

df.isnull().sum()

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

There are 309 missing values in the coupon code feature

In [3]:
# drop the missing values in the couponcode column

df= df.dropna(subset = ['CouponCode'])

In [7]:
# check for duplicates

df.duplicated().sum()

np.int64(0)

There are no duplicates in the dataset

In [8]:
# check the total number of observations(rows) after dropping the missing values in the coupon code

df.shape

(891, 14)

There are now total number of 891 rows 14 columns after dropping the missing value

In [9]:
# Checking invalid vaues

df[df['Quantity'] <= 0]

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice


In [10]:
# Verify if the total price corresponds with the unit price and quantity

CalculateTotal = df['Quantity'] * df['UnitPrice']
print(CalculateTotal)

0       2853.10
1        302.70
2       2753.40
3        273.19
4       2504.04
         ...   
1194     568.32
1195     107.04
1197     873.68
1198    1050.08
1199    2242.32
Length: 891, dtype: float64


The total price correspond with the multiplication of quantity and unit price

### Inspect the data to see if there are inconsistent format using value count


In [11]:
# check the value counts of the dataset

df['Date'].value_counts()

Date
2023-08-20    6
2024-06-08    5
2024-04-07    5
2025-03-15    4
2023-10-16    4
             ..
2023-08-04    1
2024-04-27    1
2023-12-21    1
2024-09-27    1
2024-08-22    1
Name: count, Length: 571, dtype: int64

In [12]:
df['CustomerID'].value_counts()

CustomerID
C56969    2
C46651    2
C21191    2
C97593    2
C91155    2
         ..
C28893    1
C97774    1
C11657    1
C85993    1
C57502    1
Name: count, Length: 886, dtype: int64

In [13]:
df['Product'].value_counts()

Product
Printer    139
Tablet     133
Chair      130
Desk       129
Laptop     127
Phone      117
Monitor    116
Name: count, dtype: int64

In [14]:
df['Quantity'].value_counts()

Quantity
1    200
4    186
2    177
3    172
5    156
Name: count, dtype: int64

In [15]:
df['UnitPrice'].value_counts()

UnitPrice
136.84    2
127.18    2
228.91    2
133.39    2
183.67    2
         ..
451.97    1
66.10     1
547.32    1
412.53    1
560.58    1
Name: count, Length: 886, dtype: int64

In [16]:
df['TrackingNumber'].value_counts()

TrackingNumber
TRK37947903    1
TRK88437123    1
TRK78924436    1
TRK99468332    1
TRK58443593    1
              ..
TRK50284486    1
TRK72547640    1
TRK31140158    1
TRK12320793    1
TRK51116746    1
Name: count, Length: 891, dtype: int64

In [17]:
df['PaymentMethod'].value_counts()

PaymentMethod
Cash           183
Gift Card      181
Credit Card    180
Online         178
Debit Card     169
Name: count, dtype: int64

In [18]:
df['CouponCode'].value_counts()

CouponCode
FREESHIP    313
WINTER15    292
SAVE10      286
Name: count, dtype: int64

In [19]:
df['ShippingAddress'].value_counts()

ShippingAddress
740 Main St    5
146 Main St    5
696 Main St    5
636 Main St    5
895 Main St    5
              ..
103 Main St    1
556 Main St    1
211 Main St    1
764 Main St    1
201 Main St    1
Name: count, Length: 556, dtype: int64

In [20]:
df['ItemsInCart'].value_counts()

ItemsInCart
6     138
5     135
4     125
7     119
8      94
3      91
9      64
2      59
1      39
10     27
Name: count, dtype: int64

In [21]:
df['OrderStatus'].value_counts()

OrderStatus
Cancelled    192
Delivered    179
Pending      178
Shipped      171
Returned     171
Name: count, dtype: int64

In [22]:
df['ReferralSource'].value_counts()

ReferralSource
Instagram    198
Email        195
Google       175
Referral     162
Facebook     161
Name: count, dtype: int64

There are no inconsistent format in the dataset and all rows and columns are standardized. The data is accurate and ready for exploring since the missing value in the coupon code has been dropped.